### PR TITLE
Fixes #988

### LINKED ISSUE #887: Command block cart will get a name of @ after quitting and joining a save game
Before quitting:

After rejoining:

### LINKED ISSUE #988: Under Cutter Cart Crashing Server
[crash-2016-11-01_18.35.30-server.txt](https://github.com/CovertJaguar/Railcraft/files/565243/crash-2016-11-01_18.35.30-server.txt)
When trying to add materials to the under cutter cart . my server crashes

### PR DESCRIPTION
**Pull request by liach**
Indeed I make fault sometimes.

### CODE REVIEW

File: src/main/java/mods/railcraft/common/carts/EntityCartCommand.java [hunk 1]
```diff
@@ -44,9 +44,7 @@ protected void writeEntityToNBT(NBTTagCompound compound) {
         super.writeEntityToNBT(compound);
         // Command block logic overrode this and the custom name is
         // somewhat set to the command block's name ("@")
-        if (!getCustomNameTag().isEmpty()) {
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/carts/EntityCartCommand.java:47]: hasCustomNameTag()?

**Comment by CovertJaguar** [comment]
Remember when I said not to combine multiple fixes into a single commit?

File: src/main/java/mods/railcraft/common/carts/EntityCartCommand.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/carts/EntityCartCommand.java:47]: Actually, I found that even if it does not have a custom name, the cart's name would be set to the command block's default name `@`.

**Comment by liach** [comment]
Wait a second

I will just fix #988

**Comment by liach** [comment]
See #991 for the command block minecart issue.

**Pull request merged**